### ISSUE TITLE
bootCamp

### ISSUE THREAD

**Issue by gaeSeung1** [gaeSeung1/bootCamp#1]
handson

**Comment by gaeSeung1** [comment]
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.datasets import load_boston, load_iris
from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.metrics import mean_squared_error as MSE
from sklearn import tree

**Comment by gaeSeung1** [comment]
gg

**Comment by gaeSeung1** [comment]
[1.bootcamp_2022_introduction.zip](https://github.com/gaeSeung1/bootCamp/files/9795985/1.bootcamp_2022_introduction.zip)

**Comment by gaeSeung1** [comment]
dddd

**Comment by gaeSeung** [comment]
sdfsadfsdf

**Comment by gaeSeung1** [comment]
[221017.zip](https://github.com/gaeSeung1/bootCamp/files/9796949/221017.zip)

**Comment by gaeSeung1** [comment]
from detecta import detect_peaks
data12 = data12 #여기에 데이터프레임 파일 입력

data12_NG = data12.iloc[:,4:].loc[data12['label']==1]
data12_OK = data12.iloc[:,4:].loc[data12['label']==0]

ref = 50
for i in range(100):
    df_peak_idx = data12_OK.iloc[i,:].loc[data12_OK.iloc[i,:]>=ref]
    peak_idx = detect_peaks(df_peak_idx)
    peak = df_peak_idx.iloc[peak_idx]
    plt.scatter(peak.index,peak,s=15,c='b')

for i in range(60):
    df_peak_idx = data12_NG.iloc[i,:].loc[data12_NG.iloc[i,:]>=ref]
    peak_idx = detect_peaks(df_peak_idx)
    peak = df_peak_idx.iloc[peak_idx]
    plt.scatter(peak.index,peak,s=15,c='r')

**Comment by gaeSeung1** [comment]
#양품 vs 고품 그래프
for j in range(1,3):
    plt.subplot(1,2,j)
    for i in range(100):
        try:
            if j == 1:
                plt.plot(data12_OK.iloc[i,4:])
            else:
                plt.plot(data12_NG.iloc[i,4:])
        except:
            None
    plt.ylim(0,500)

**Comment by kimnko** [comment]
raw_data.groupby(["date","item_code"])["label"].value_counts()

**Comment by gaeSeung1** [comment]
data12 = data12 #데이터 프레임 입력
data12_NG = data12.iloc[:,4:].loc[data12['label']==1]
data12_OK = data12.iloc[:,4:].loc[data12['label']==0]

lst = []
for i in range(10063):
    df_peak_OK_idx = data12_OK.iloc[i,:].loc[data12_OK.iloc[i,:]>=ref]
    peak_OK_idx = detect_peaks(df_peak_OK_idx)
    peak_OK = df_peak_OK_idx.iloc[peak_OK_idx]
    lst.append(peak_OK.mean())
lst
data12_OK.index
plt.scatter(data12_OK.index,lst,s=5)

**Comment by kimnko** [comment]
import scipy.stats as stats

#정규성검증
cols = group1.columns
norm_group1 = []
norm_group2 = []

for i in range(len(cols)):
    x1 = group1.iloc[:,i]
    x2 = group2.iloc[:,i]
    t_stat_x1, p_val_x1 = stats.kstest(x1, 'norm', args=(x1.mean(), x1.var()**0.5))
    t_stat_x2, p_val_x2 = stats.kstest(x2, 'norm', args=(x2.mean(), x2.var()**0.5))

    if p_val_x1 < 0.05:
        norm_group1.append(cols[i])
    if p_val_x1 < 0.05:
        norm_group2.append(cols[i])

#집단간 분산 검증
cols = group1.columns
bar_groups = []

for i in range(len(cols)):
    x1 = group1.iloc[:,i]
    x2 = group2.iloc[:,i]
    var_test_x1x2, var_test_p_val_x1x2 = stats.bartlett(x1, x2)

    if var_test_p_val_x1x2 > 0.05:
        bar_groups.append(cols[i])

**Comment by chae13** [comment]
# data = 원본
# Standard scale
data_std1 = data.iloc[:,4:]

scaler = StandardScaler()
data_std2 = scaler.fit_transform(data_std1)
data_scaled = pd.DataFrame(data_std2, columns = data_std1.columns)

# PCA 라이브러리
from sklearn.decomposition import PCA

pca = PCA(n_components=100)
pca_array = pca.fit_transform(data_scaled)
pca_df = pd.DataFrame(pca_array, index=data_scaled.index,
                      columns=[f"pca{num+1}" for num in range(100)])
pca_df.head()

# PCA 주성분개수 결정 고유값 0.7 이상, 누적기여율 80%이상
pd.set_option('display.max_rows', None)

result = pd.DataFrame({'설명가능한 분산 비율(고윳값)':pca.explained_variance_,
             '기여율':pca.explained_variance_ratio_},
            index=np.array([f"pca{num+1}" for num in range(100)]))
result['누적기여율'] = result['기여율'].cumsum()
result

# Scree plot 엘보우 분석
plt.title('scree plot')
plt.xlabel('result.index')
plt.ylabel('고유값')
plt.plot(pca.explained_variance_,'o-')
plt.xlim(0,25)

**Comment by gaeSeung** [comment]
[rng.csv](https://github.com/gaeSeung1/bootCamp/files/9805792/rng.csv)

**Comment by gaeSeung1** [comment]
[rng_p.csv](https://github.com/gaeSeung1/bootCamp/files/9806341/rng_p.csv)

**Comment by kimnko** [comment]
from sklearn.metrics import confusion_matrix
confusion_matrix(Test_Y, y_pred)

from sklearn.metrics import classification_report
print(classification_report(Test_Y, y_pred))

**Comment by gaeSeung1** [comment]
df = pd.DataFrame()
data12_freq = data12.iloc[:,4:]
ref = 2500
for i in range(8):
    df1 = data12_freq.loc[:,(data12_freq.columns >= ref * i) & (data12_freq.columns < ref + ref * i)]
    df['amp%d'%(i+1)] = df1.max(axis=1)
    df['freq%d'%(i+1)] = df1.idxmax(axis=1)
df

**Comment by chae13** [comment]
ser1 = data[data['label']==1].index.to_series()

for i in ser1:
    plt.plot(data.iloc[i,4:])

plt.show()

**Comment by gaeSeung1** [comment]
def peak(data12):
        ref = 50

        data12_NG = data12.loc[data12['label']==1]
        data12_OK = data12.loc[data12['label']==0]

        df_peak_OK = pd.DataFrame()
        df_peak_NG = pd.DataFrame()

        for i in range(len(data12_OK)):
            df_peak_OK_idx = data12_OK.iloc[i,4:].loc[data12_OK.iloc[i,4:]>=ref]
            peak_OK_idx = detect_peaks(df_peak_OK_idx)
            peak_OK = pd.DataFrame(df_peak_OK_idx.iloc[peak_OK_idx])
            peak_OK['row'] = i
            peak_OK.reset_index(inplace=True)
            peak_OK.columns = ['freq','amp','row']
            df_peak_OK = pd.concat([df_peak_OK, peak_OK],ignore_index=True)

        df_peak_OK['label'] = 'OK'

        for i in range(len(data12_NG)):
            df_peak_NG_idx = data12_NG.iloc[i,4:].loc[data12_NG.iloc[i,4:]>=ref]
            peak_NG_idx = detect_peaks(df_peak_NG_idx)
            peak_NG = pd.DataFrame(df_peak_NG_idx.iloc[peak_NG_idx])
            peak_NG['row'] = i
            peak_NG.reset_index(inplace=True)
            peak_NG.columns = ['freq','amp','row']
            df_peak_NG = pd.concat([df_peak_NG, peak_NG],ignore_index=True)

        df_peak_NG['label'] = 'NG'

        df_peak = pd.concat([df_peak_OK,df_peak_NG],ignore_index=True)
        return df_peak

    #data12에 데이터프레임 입력
    df_peak = peak(data12)


    import plotly.express as px
    fig = px.scatter(df_peak, x="freq", y="amp", color="label", symbol="label")
    fig.show()
